models/RwdSet.php

```diff
@@ -88,12 +96,40 @@ public function parse_options( $params ) {
 				}
 			} else {
 				$nested_key               = ( $this->key === $subkey ) ? $this->key : "$this->key-$subkey";
-				$this->options[ $subkey ] = new RwdOption( $nested_key, $conf );
+				$this->options[ $subkey ] = new RwdOption( $nested_key, $conf, $retina );
 			}
 		}
 
 		// get first option and take size object to save it as Set size.
 		$first      = reset( $this->options );
-		$this->size = new ImageSize( $this->key, array( $first->size->w, $first->size->h, $first->size->crop ) );
+		$this->size = new ImageSize( $this->key, array( $first->size->w, $first->size->h, $first->size->crop ), $retina );
+	}
+
+	/**
+	 * Parse key to create Retina array options size.
+	 *
+	 * @param string $key Base image size key.
+	 */
+	public function parse_retina_options( $key ) {
+		preg_match_all( '/(\s[0-9.]+x)/', $key, $retina_parse );
+		if( $retina_parse ){
+			foreach( $retina_parse as $key_retina => $retina_value ) {
+				foreach( $retina_value as $value ){
+					$this->retina[ trim($value) ] = floatval( $value );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Clean image size key without retina size.
+	 *
+	 * @param string $key Base image size key.
+	 *
+	 * @return string Image size key.
+	 */
+	public function clean_key( $key ) {
+		preg_match('/^(\s?\S+)/', $key, $clean_key);
+		return trim( $clean_key[0] );
```

Comment:
there is preg_replace function to replace what you need.
this is not correct, because I can set name like 'home related 2x'. and in this case key will be only 'home'.
You should remove THE SAME code you get with regular expression for retina descriptors